default/AI/ShipDesignAI.py

```diff
@@ -462,6 +468,8 @@ def _update_buildable_items_this_turn(self, verbose=False):
             if res:
                 if verbose:
                     print "Success: Added Test Design %s, with result %d" % (testdesign_name, res)
+                # update cache since we just created a new design in fo
+                _ = _get_design_by_name(testdesign_name,update_invalid=True)
```

Comment:
Before we have lazy cache initialization. We need design, we create design in cache.
Now we add design on creation, am I right? May be split get and set in that case?